### PR TITLE
[CLOUDTRUST-4175] missing apis in bridge

### PR DESCRIPTION
**Pull request by basbeu**

### CODE REVIEW

File: api/management/swagger-api_management.yaml [hunk 1]
```diff
@@ -1827,6 +1858,25 @@ paths:
...
+        in: path
+        description: realm name (not id!)
+        required: true
+        schema:
+          type: string
+      responses:
+        200:
+          description: Login provider id
+          content:
+            application/json:
```
Comment (harture) [inline api/management/swagger-api_management.yaml:1877]: Detail, you could define the response as an array of IdentityProvider.

File: go.mod [hunk 2]
```diff
@@ -5,7 +5,7 @@ go 1.17
 require (
 	github.com/cloudtrust/common-service/v2 v2.5.6
 	github.com/cloudtrust/httpclient v1.0.7
-	github.com/cloudtrust/keycloak-client/v2 v2.5.5
+	github.com/cloudtrust/keycloak-client/v2 v2.5.6-0.20220506080911-67c249efcf92
```
Comment (harture) [inline go.mod:8]: seems to be a branch, do not forget to create a release and pointing on it

**Review by harture — APPROVED** [review]

**Review by harture — COMMENTED** [review]
Maybe change the commit message with what you are adding

File: api/management/swagger-api_management.yaml (hunk 1 above)
Comment (basbeu) [reply api/management/swagger-api_management.yaml:1877]: Done

File: api/management/swagger-api_management.yaml [hunk 3]
```diff
@@ -1827,6 +1858,25 @@ paths:
       responses:
         200:
           description: successful operation  
+  /realms/{realm}/identity-provider/instances:
```
[the remark this answers is not in the document]
Comment (basbeu) [reply api/management/swagger-api_management.yaml:1861]: it was just matching the keycloak API. I changed it to  /realms/{realm}/identity-providers

File: pkg/management/authorization.go [hunk 4]
```diff
@@ -51,6 +51,7 @@ var (
 	MGMTSetTrustIDGroups                    = newAction("MGMT_SetTrustIDGroups", security.ScopeGroup)
 	MGMTGetClientRolesForUser               = newAction("MGMT_GetClientRolesForUser", security.ScopeGroup)
 	MGMTAddClientRolesToUser                = newAction("MGMT_AddClientRolesToUser", security.ScopeGroup)
+	MGMTDeleteClientRolesFromUser           = newAction("MGMT_DeleteClientRolesFromUser", security.ScopeGroup)
```
Comment (fperot74) [inline pkg/management/authorization.go:54]: Does front team have been adviced of these new actions?

File: pkg/management/component.go [hunk 5]
```diff
@@ -79,6 +80,7 @@ type KeycloakClient interface {
 	LinkShadowUser(accessToken string, realmName string, userID string, provider string, fedID kc.FederatedIdentityRepresentation) error
 	ClearUserLoginFailures(accessToken string, realmName, userID string) error
 	GetAttackDetectionStatus(accessToken string, realmName, userID string) (map[string]interface{}, error)
+	GetIdps(accessToken string, realmName string) ([]kc.IdentityProviderRepresentation, error)
```
Comment (fperot74) [inline pkg/management/component.go:83]: Why don't we name this function GetIdentityProviders as we usually have full names in our functions?

File: pkg/management/mock_test.go [hunk 6]
```diff
@@ -12,3 +12,7 @@ package management
 func ptr(value string) *string {
 	return &value
 }
+
+func boolPtr(value bool) *bool {
```
Comment (fperot74) [inline pkg/management/mock_test.go:16]: It's been a long time I have worked on Go projects but I can remember these functions are usually named ptr, ptrInt, ptrXXX, ...

**Review by fperot74 — APPROVED** [review]
Could you reword your commit with something like "Add get identity providers and delete client role functions"

File: pkg/management/mock_test.go (hunk 6 above)
Comment (basbeu) [reply pkg/management/mock_test.go:16]: Good catch

File: pkg/management/component.go (hunk 5 above)
Comment (basbeu) [reply pkg/management/component.go:83]: It is the naming in the keycloak-client

**Pull request merged**